packages/nx/src/command-line/migrate.ts

```diff
@@ -1040,7 +1040,16 @@ async function runNxMigration(root: string, packageName: string, name: string) {
 
   const collection = readJsonFile<MigrationsJson>(collectionPath);
   const g = collection.generators || collection.schematics;
-  const implRelativePath = g[name].implementation || g[name].factory;
+
+  let implRelativePath: string;
+  try {
+    const c = g[name];
+    implRelativePath = c.implementation || c.factory;
+  } catch (e) {
+    throw new Error(
+      `Can not get relative path for migration's implementation as collection not found by name="${name}" for package="${packageName}", collectionPath="${collectionPath}"`
+    );
+  }
```

Comment:
Rather than a try catch here, we could do something like this:
```suggestion
  const c = g[name];
  implRelativePath = g[name]?.implementation || g[name]?.factory
  if (!implRelativePath) {
    throw new Error(
      `Unable to determine relative path for migration implementation: { "name": "${name}", "package": "${packageName}", "collectionPath": "${collectionPath}"`
    );
  }
```